lib/notifiers/filtered_users_notifier.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:first_project/enums/connection_status.dart';
+import 'package:first_project/enums/leave_type.dart';
+import 'package:first_project/models/user.dart';
+import 'package:first_project/notifiers/all_users_notifier.dart';
+import 'package:first_project/notifiers/filters_notifier.dart';
+import 'package:first_project/notifiers/search_query_notifier.dart';
+import 'package:flutter/material.dart';
+
+class FilteredUsersNotifier extends ChangeNotifier {
+  final AllUsersNotifier allUsersNotifier;
+  final FiltersNotifier filtersNotifier;
+  final SearchQueryNotifier searchQueryNotifier;
+  List<User> _filteredUsers = [];
+
+  FilteredUsersNotifier({
+    required this.allUsersNotifier,
+    required this.filtersNotifier,
+    required this.searchQueryNotifier,
+  }) {
+    allUsersNotifier.addListener(_filterUsers);
```

Comment:
@KarloK34 Why adding listeners here? What you should do is call filteredUsersNotifier?.filterUsers(allUsers,searchQuery,filters) in the update callbacl of ChangeProxyNotifier